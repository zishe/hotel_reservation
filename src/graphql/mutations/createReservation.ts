import { gql } from "apollo-boost";

export const CREATE_RESERVATION_MUTATION = gql`
  mutation CreateReservation(
    $name: String!
    $hotelName: String!
    $arrivalDate: String!
    $departureDate: String!
  ) {
    createReservation(
      data: {
        name: $name
        hotelName: $hotelName
        arrivalDate: $arrivalDate
        departureDate: $departureDate
      }
    ) {
      id
      name
      hotelName
      arrivalDate
      departureDate
    }
  }
`;
