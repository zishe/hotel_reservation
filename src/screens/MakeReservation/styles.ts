import { StyleSheet, Dimensions } from "react-native";
import RF from "react-native-responsive-fontsize";
const width = Dimensions.get("window").width;

export default StyleSheet.create({
  contentContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  subHeader: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#bbb"
  },
  content: {
    flex: 1,
    flexDirection: "column",
    justifyContent: "center"
  },
  body: {
    flex: 9,
    justifyContent: "center"
  },
  subHeaderText: {
    fontSize: RF(3.2),
    color: "#666"
  },
  textBox: {
    marginTop: 9,
    borderColor: "#aaa",
    borderWidth: 1.5,
    padding: 5,
    borderRadius: 4,
    height: 60,
    width: width - 30,
    justifyContent: "center"
  },
  rowSpan1: {
    flex: 2,
    marginTop: 8,
    justifyContent: "center",
    width,
    padding: 15,
    alignItems: "stretch",
    borderBottomWidth: 1,
    borderColor: "#ddd"
  },
  button: {
    marginTop: 8,
    fontSize: 20,
    borderRadius: 4,
    height: 70
  },
  buttonText: {
    fontSize: RF(3.2),
    color: "#FFF"
  }
});
