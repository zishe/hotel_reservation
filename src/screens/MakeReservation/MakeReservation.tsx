import React from "react";
import { Text, TextInput, View, ActivityIndicator } from "react-native";
import {
  Container,
  Button,
  Content,
  Icon,
  DatePicker,
  Toast
} from "native-base";
import { Mutation } from "react-apollo";
import { format, addDays } from "date-fns";
import { CREATE_RESERVATION_MUTATION } from "../../graphql/mutations/createReservation";
import { AppHeader } from "../../components";
import styles from "./styles";
import { SELECTED_RESERVATION_ID } from "../../graphql/client/queries/selectedReservation";

interface Props {
  navigation: any;
}

interface State {
  name: string;
  hotelName: string;
  arrivalDate: string | Date;
  departureDate: string | Date;
}

interface FormField {
  name: string;
  label: string;
  type: "text" | "date";
}

const formFields: FormField[] = [
  { name: "name", label: "Guest Name", type: "text" },
  { name: "hotelName", label: "Hotel Name", type: "text" },
  { name: "arrivalDate", label: "Arrival Date", type: "date" },
  { name: "departureDate", label: "Departure Date", type: "date" }
];

/**
 * @export
 * @class MakeReservation
 * @extends {React.PureComponent<Props, State>}
 */
export default class MakeReservation extends React.PureComponent<Props, State> {
  static navigationOptions = {
    drawerLabel: "Make New Reservation",
    drawerIcon: (): any => <Icon name="add" />
  };

  state: State = {
    name: "",
    hotelName: "",
    arrivalDate: new Date(),
    departureDate: addDays(new Date(), 1)
  };

  screenFocusSubscription: any = null;

  // subscribe to 'didFocus' event of react-navigation
  // whenever user lands on this screen, set the initial state
  subscribeToDidFocus = () => {
    this.screenFocusSubscription = this.props.navigation.addListener(
      "didFocus",
      () => {
        this.initializeState();
      }
    );
  };

  // subscribe didFocus event
  componentWillMount() {
    this.subscribeToDidFocus();
  }

  // unsubscribe(remove) didFocus event
  componentWillUnmount() {
    this.screenFocusSubscription.remove();
  }

  /**
   * @name initializeState
   */
  initializeState = () => {
    this.setState({
      name: "",
      hotelName: "",
      arrivalDate: new Date(),
      departureDate: addDays(new Date(), 1)
    });
  };

  /**
   * @name render
   * @desc component render method
   */
  render() {
    return (
      <Mutation
        mutation={CREATE_RESERVATION_MUTATION}
        variables={{ ...this.state }}
        update={this.updateStore}
        refetchQueries={[`GetBookingsQuery`]}
        awaitRefetchQueries={true}
      >
        {(createReservation, { loading, error }) =>
          this.renderForm(createReservation, loading, error)
        }
      </Mutation>
    );
  }

  /**
   * @name renderForm
   * @desc renders the form for making a new reservation
   */
  renderForm = (createReservation: any, loading: boolean, error: any) => (
    <Container>
      <AppHeader
        title="Hilton Reservations"
        navigation={this.props.navigation}
      />
      <View style={styles.content}>
        {error && this.onReservationError(error)}
        <View style={styles.subHeader}>
          <Text style={styles.subHeaderText}>Make New Reservation</Text>
        </View>

        <View style={styles.body}>
          <Content padder contentContainerStyle={styles.contentContainer}>
            {formFields.map(field => this.renderField(field))}
            {this.renderSubmitButton(createReservation, loading)}
          </Content>
        </View>
      </View>
    </Container>
  );

  /**
   * @name renderField
   * @param field: FormField Field object of the element to be rendered
   * @description renders a field of MakeReservation form
   * @memberof MakeReservation
   */
  renderField = (field: FormField) => {
    return (
      <View style={[styles.rowSpan1]} key={field.label}>
        <View>
          <Text>{field.label}</Text>
        </View>
        <View>
          {field.type === "date"
            ? this.renderDatePicker(field)
            : this.renderTextInput(field)}
        </View>
      </View>
    );
  };

  /**
   * @name renderTextInput
   * @param field: FormField Field object of the element to be rendered
   * @description renders a TextInput type field of MakeReservation form
   * @memberof MakeReservation
   */
  renderTextInput = (field: FormField) => {
    return (
      <TextInput
        testID={field.name}
        style={styles.textBox}
        placeholder={field.label}
        value={(this.state as any)[field.name]}
        onChangeText={txt => this.onTextChange(field.name, txt)}
      />
    );
  };

  /**
   * @name renderDatePicker
   * @param field: FormField Field object of the element to be rendered
   * @description renders a DatePicker element
   * @memberof MakeReservation
   */
  renderDatePicker = (field: FormField) => {
    // set default value for arrivalDate=today, departure date = tomorrow
    const defaultDate =
      field.name === "departureDate" ? addDays(new Date(), 1) : new Date();

    return (
      <View style={styles.textBox}>
        <DatePicker
          defaultDate={defaultDate}
          minimumDate={defaultDate}
          maximumDate={new Date(2999, 12, 31)}
          modalTransparent={false}
          animationType={"fade"}
          androidMode={"default"}
          textStyle={{ color: "green" }}
          placeHolderTextStyle={{ color: "#d3d3d3" }}
          testID={field.name}
          onDateChange={date => this.datePicked(date, field.name)}
          formatChosenDate={(date: string) => format(date, "MM/DD/YYYY")}
          disabled={false}
        />
      </View>
    );
  };

  /**
   * @name datePicked
   * @param date the date returned from DatePicker
   * @description sets the state.arrivalDate to selected date
   * @memberof MakeReservation
   */
  datePicked = (date: string, fieldName: string) =>
    this.setState({ [fieldName]: date } as any);

  /**
   * @name renderSubmitButton
   * @description render Submit button, show a loading spinner with disabled status when query is running
   * @memberof MakeReservation
   */
  renderSubmitButton = (createReservation: any, loading: boolean) => {
    return (
      <View style={[styles.rowSpan1]}>
        <Button
          iconLeft
          full
          style={[styles.button]}
          disabled={loading}
          onPress={createReservation}
          testID="SubmitButton"
        >
          {loading && (
            <ActivityIndicator
              size="small"
              color="#00ff00"
              testID="activityIndicator"
            />
          )}
          <Text style={styles.buttonText}>Make Reservation</Text>
        </Button>
      </View>
    );
  };

  /**
   * @name onReservationError
   * @description callback to show Mutation error
   * @memberof MakeReservation
   */
  onReservationError = (error: any) => {
    this.toastMessage(
      error.message || "Server or Network error, please try again",
      "danger"
    );
  };

  /**
   * @name onTextChange
   * @description function to call onChangeText event of text inputs
   *              to update the corresponding state value
   * @param fieldName name of the input whose text value has been changed
   * @param value value of the input field
   * @memberof MakeReservation
   */
  onTextChange = (fieldName: string, value: string) => {
    this.setState({
      [fieldName]: value.trim()
    } as any);
  };

  /**
   * @name toastMessage
   * @param msg Message to show on toast
   * @memberof MakeReservation
   */
  toastMessage = (message: string, type: any = "success") => {
    return Toast.show({
      text: message,
      type: type,
      buttonText: "Okay",
      position: "top",
      duration: 5000
    });
  };

  /**
   * @name onReservationComplete
   * @description callback for Mutation onComplete
   * @memberof MakeReservation
   */
  onReservationComplete = () => {
    this.toastMessage("Congratulations! Room reserved successfully!");
    this.props.navigation.navigate("Bookings");
  };

  /**
   * @name updateStore
   * @param store - ApolloClient Cache store
   * @param mutationResult - result of the mutation
   * @description, while ApolloClient would auto-update the store cache of reservations
   *               we need to manually update the selectedReservationId to newly added
   *               reservation to higlight it in the list page
   * @memberof MakeReservation
   */
  updateStore = async (store: any, mutationResult: any) => {
    const {
      data: { createReservation }
    } = mutationResult;

    await store.writeQuery({
      query: SELECTED_RESERVATION_ID.query,
      data: { selectedReservationId: createReservation.id }
    });

    this.onReservationComplete();
  };
}
