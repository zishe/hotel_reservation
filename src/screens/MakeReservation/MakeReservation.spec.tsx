import React from "react";
import { MockedProvider } from "react-apollo/test-utils";
import renderer from "react-test-renderer";

import MakeReservation from "./MakeReservation";
import { CREATE_RESERVATION_MUTATION } from "../../graphql/mutations/createReservation";

const mockProps = (props: Object) => ({
  navigation: {
    navigate: jest.fn(),
    addListener: jest.fn()
  },
  ...props
});

const mockReservation = {
  name: "TestName",
  arrivalDate: "03/20/2019",
  hotelName: "TestHotelName",
  departureDate: "03/22/2019"
};

// mock query for valid data/response
const mockReservationMutation = {
  request: {
    query: CREATE_RESERVATION_MUTATION,
    variables: { ...mockReservation }
  },
  result: {
    data: {
      createReservation: { ...mockReservation, id: "Test-1" }
    }
  }
};

// mock query for error response
// const mockReservationMutationError: any = {
//   request: {
//     mutation: CREATE_RESERVATION_MUTATION
//   },
//   error: {
//     message: "TestError"
//   }
// };
jest.useFakeTimers();
describe("MakeReservationScreen", () => {
  let props: any;
  beforeEach(() => {
    props = mockProps({});
  });

  /**
   * @description test component renders
   */
  it("should render as expected", () => {
    const bookings = renderer
      .create(
        <MockedProvider mocks={[]}>
          <MakeReservation {...props} />
        </MockedProvider>
      )
      .toJSON();
    expect(bookings).toMatchSnapshot();
  });

  /**
   * @description test for valid data
   */
  it("should show ActivityIndicator on press of submit", async () => {
    const component = renderer.create(
      <MockedProvider mocks={[mockReservationMutation]} addTypename={false}>
        <MakeReservation {...props} />
      </MockedProvider>
    );

    const submitButton = component.root.find(
      el => el.props.testID === "SubmitButton"
    );
    submitButton.props.onPress();

    const activityIndicator = component.root.findAll(
      (el: any) =>
        el.type === "ActivityIndicator" &&
        el.props.testID === "activityIndicator"
    );
    expect(activityIndicator).toHaveLength(1);
  });

  /**
   * @description test for valid data
   * @seems some issue with the Mutation unit testing
   * GitHub issue to be logged
   * https://medium.com/hotel-reservation-app/initialize-project-c19f57ff2e7e/#fbab
   */
  //   it("should call createReservation mutation on press of submit", async () => {
  //     props = mockProps({});
  //     const component = renderer.create(
  //       <MockedProvider mocks={[mockReservationMutation]} addTypename={false}>
  //         <MakeReservationScreen {...props} />
  //       </MockedProvider>
  //     );

  //     // const instance = component.root.instance;

  //     // instance.setState({
  //     //   name: "1",
  //     //   hotelName: "1",
  //     //   arrivalDate: "03/21/2019",
  //     //   departureDate: "03/22/2019"
  //     // });
  //     // component.root
  //     //   .find(el => el.props.testID === "arrivalDate")
  //     //   .props.onDateChange(mockReservation.arrivalDate);

  //     await wait(0); // wait for response
  //     expect(props.navigation.navigate).toBeCalled();
  //   });
});
