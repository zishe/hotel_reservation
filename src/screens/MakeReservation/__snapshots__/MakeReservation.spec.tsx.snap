// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MakeReservationScreen should render as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      androidStatusBarColor="lightgray"
      iosBarStyle="dark-content"
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingTop": 18,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          },
          Object {
            "backgroundColor": "#104c97",
            "elevation": 4,
            "shadowColor": "black",
            "shadowOpacity": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            },
            Object {
              "flex": 2,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#fff",
                  "padding": 5,
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="menu"
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
            },
            Object {
              "alignItems": "center",
              "flex": 4,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "700",
                "marginLeft": undefined,
                "paddingLeft": 4,
                "paddingTop": 1,
                "textAlign": "center",
              },
              Object {
                "color": "#fff",
              },
            ]
          }
        >
          Hilton Reservations
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            },
            Object {
              "flex": 2,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#fff",
                  "padding": 5,
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="contact"
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#bbb",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#666",
            "fontSize": 43,
          }
        }
      >
        Make New Reservation
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 9,
          "justifyContent": "center",
        }
      }
    >
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={
          Array [
            Object {
              "padding": 10,
            },
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutomaticScroll={true}
        enableOnAndroid={false}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        padder={true}
        resetKeyboardSpace={[Function]}
        resetScrollToCoords={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollIntoView={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          }
        }
        update={[Function]}
        viewIsInsideTabBar={false}
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "stretch",
                  "borderBottomWidth": 1,
                  "borderColor": "#ddd",
                  "flex": 2,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "padding": 15,
                  "width": 750,
                },
              ]
            }
          >
            <View>
              <Text>
                Guest Name
              </Text>
            </View>
            <View>
              <TextInput
                autoFocus={false}
                blur={false}
                onChangeText={[Function]}
                placeholder="Guest Name"
                style={
                  Object {
                    "borderColor": "#aaa",
                    "borderRadius": 4,
                    "borderWidth": 1.5,
                    "height": 60,
                    "justifyContent": "center",
                    "marginTop": 9,
                    "padding": 5,
                    "width": 720,
                  }
                }
                testID="name"
                value=""
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "stretch",
                  "borderBottomWidth": 1,
                  "borderColor": "#ddd",
                  "flex": 2,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "padding": 15,
                  "width": 750,
                },
              ]
            }
          >
            <View>
              <Text>
                Hotel Name
              </Text>
            </View>
            <View>
              <TextInput
                autoFocus={false}
                blur={false}
                onChangeText={[Function]}
                placeholder="Hotel Name"
                style={
                  Object {
                    "borderColor": "#aaa",
                    "borderRadius": 4,
                    "borderWidth": 1.5,
                    "height": 60,
                    "justifyContent": "center",
                    "marginTop": 9,
                    "padding": 5,
                    "width": 720,
                  }
                }
                testID="hotelName"
                value=""
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "stretch",
                  "borderBottomWidth": 1,
                  "borderColor": "#ddd",
                  "flex": 2,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "padding": 15,
                  "width": 750,
                },
              ]
            }
          >
            <View>
              <Text>
                Arrival Date
              </Text>
            </View>
            <View>
              <View
                style={
                  Object {
                    "borderColor": "#aaa",
                    "borderRadius": 4,
                    "borderWidth": 1.5,
                    "height": 60,
                    "justifyContent": "center",
                    "marginTop": 9,
                    "padding": 5,
                    "width": 720,
                  }
                }
              >
                <View>
                  <View>
                    <Text
                      onPress={[Function]}
                      style={
                        Array [
                          Object {
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          },
                          Object {
                            "color": "#000",
                            "padding": 10,
                          },
                          Object {
                            "color": "green",
                          },
                        ]
                      }
                      uppercase={false}
                    >
                      03/26/2019
                    </Text>
                    <View>
                      <Modal
                        animationType="fade"
                        hardwareAccelerated={false}
                        onRequestClose={[Function]}
                        supportedOrientations={
                          Array [
                            "portrait",
                            "landscape",
                          ]
                        }
                        transparent={false}
                        visible={false}
                      >
                        <Text
                          onPress={[Function]}
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "backgroundColor": "transparent",
                                "flex": 1,
                              },
                            ]
                          }
                          uppercase={false}
                        />
                        <DatePickerIOS
                          date={2019-03-26T13:02:48.511Z}
                          maximumDate={3000-01-31T06:00:00.000Z}
                          minimumDate={2019-03-26T13:02:48.511Z}
                          mode="date"
                          onDateChange={[Function]}
                          setNativeProps={false}
                        />
                      </Modal>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "stretch",
                  "borderBottomWidth": 1,
                  "borderColor": "#ddd",
                  "flex": 2,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "padding": 15,
                  "width": 750,
                },
              ]
            }
          >
            <View>
              <Text>
                Departure Date
              </Text>
            </View>
            <View>
              <View
                style={
                  Object {
                    "borderColor": "#aaa",
                    "borderRadius": 4,
                    "borderWidth": 1.5,
                    "height": 60,
                    "justifyContent": "center",
                    "marginTop": 9,
                    "padding": 5,
                    "width": 720,
                  }
                }
              >
                <View>
                  <View>
                    <Text
                      onPress={[Function]}
                      style={
                        Array [
                          Object {
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          },
                          Object {
                            "color": "#000",
                            "padding": 10,
                          },
                          Object {
                            "color": "green",
                          },
                        ]
                      }
                      uppercase={false}
                    >
                      03/27/2019
                    </Text>
                    <View>
                      <Modal
                        animationType="fade"
                        hardwareAccelerated={false}
                        onRequestClose={[Function]}
                        supportedOrientations={
                          Array [
                            "portrait",
                            "landscape",
                          ]
                        }
                        transparent={false}
                        visible={false}
                      >
                        <Text
                          onPress={[Function]}
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "backgroundColor": "transparent",
                                "flex": 1,
                              },
                            ]
                          }
                          uppercase={false}
                        />
                        <DatePickerIOS
                          date={2019-03-27T13:02:48.512Z}
                          maximumDate={3000-01-31T06:00:00.000Z}
                          minimumDate={2019-03-27T13:02:48.512Z}
                          mode="date"
                          onDateChange={[Function]}
                          setNativeProps={false}
                        />
                      </Modal>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "stretch",
                  "borderBottomWidth": 1,
                  "borderColor": "#ddd",
                  "flex": 2,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "padding": 15,
                  "width": 750,
                },
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#007aff",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 4,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 2,
                  "flexDirection": "row",
                  "fontSize": 20,
                  "height": 70,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                }
              }
              testID="SubmitButton"
            >
              <Text
                style={
                  Object {
                    "color": "#FFF",
                    "fontSize": 43,
                  }
                }
              >
                Make Reservation
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
